lib/wallet/wallet.js

```diff
@@ -1197,10 +1197,11 @@ class Wallet extends EventEmitter {
    * @param {Buffer} nameHash
    * @param {Address} address
    * @param {Amount} value
-   * @returns {Buffer}
+   * @param {boolean} forAllKeys
+   * @returns {Buffer|Buffer[]}
    */
 
-  async generateNonce(nameHash, address, value) {
+  async generateNonce(nameHash, address, value, forAllKeys) {
```

Comment:
Yeah I didn't like it either. Now it has `generateNonce(s)` and `generateBlind(s)`. But `generateNonce` cannot call its plural method [0] because the order of the nonces/blinds is different:
- hsd behavior has been own public key everywhere
- now we switch to smallest public key

So generateNonce has the new style (smallest) whereas generateNonces returns first based on own key so that there's no breaking change.

>The http call is kinda funny because it doesn't add anything to db ... That would make this PR a breaking change, though.

I don't think Bob uses that remote call since it always has access to the wallet directly. But since it's breaking, can be kept aside for a different PR

Reply:
Also, each commit here undos a lot of code from the previous commit so might be easier to just look at overall PR changes. Will squash if the code makes sense now.